aws_lambda_powertools/utilities/batch/base.py

```diff
@@ -100,6 +101,26 @@ def process(self) -> List[Tuple]:
         """
         return [self._process_record(record) for record in self.records]
 
+    @abstractmethod
+    async def _async_process_record(self, record: dict):
+        """
+        Async process record with handler.
+        """
+        raise NotImplementedError()
+
+    def async_process(self) -> List[Tuple]:
+        """
+        Async call instance's handler for each record.
```

Comment:
For future maintenance, let's add a comment on the use of async closure to ease decorator chaining